ext/spl/spl_iterators.stub.php

```diff
@@ -109,7 +109,7 @@ public function getMaxDepth() {}
 
 interface OuterIterator extends Iterator
 {
-    /** @return Iterator */
+    /** @return Iterator|null */
```

Comment:
Did you make this change due to the `IteratorIterator::getInnerIterator()` return type? I don't think that one was correct and have adjusted it in https://github.com/php/php-src/commit/c252420d0842b1f7ea3891695bb6de3c9ec9a57c.

Reply:
Yeah, that was the reason. I'll revert this change then

Reply:
Oops, looks like my new assertion is failing on the Zend/tests/arginfo_zpp_mismatch.phpt test. Maybe my analysis wasn't right... will have to look closer.

Reply:
OK, I'll hold on :)

Reply:
The problematic case is this:
```
$it = new AppendIterator();
var_dump($it->getInnerIterator());
```
For AppendIterator the "inner iterator" is the current iterator that AppendIterator is using, which is none here.

I'm not quite sure whether that method should be throwing, or whether the interface should really only require `?object` as the return type.

Reply:
In the interest of keeping things moving, let's go back to your original version.

Reply:
Yeah, I agree. I've just got rid of my latest commit